### PR TITLE
Add Select all button into the control

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
Add optional `Select all` button

* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility
  * [x] RTL
     * [x] When Clear button is visible/unvisible/
     * [x] When Select all button is  visible/unvisible and Clear button dynamicly appears
  * [x] Responsive
* Security, devops, perf: N/A
* UX and docs: N/A

### Demo 
select-all-demo

### CODE REVIEW

File: tree-filter.js [hunk 1]
```diff
@@ -216,6 +216,10 @@ export class Tree {
 		this._state.clear();
 	}
 
+	selectAll() {
+		this.selected = this.ids;
```
Comment (johngwilkinson) [inline tree-filter.js:220]: With a large tree, is this inefficient? Looks like it may select descendants multiple times, but I'm not sure.

Comment (Vitalii-Misechko) [reply tree-filter.js:220]: Good point!
`selected` setter contains `clearAll` step which is not needed. 
`select descendants multiple times` is also a possible room for improvements.
From the getter side `ids` all is optimal as possible (reading keys of the map).

Comment (Vitalii-Misechko) [reply tree-filter.js]: There is an effective fix thank you for pointing out
```JS
	selectAll() {
		this.setSelected(this.rootId, true);
	}
```

**Review by NicholasHallman — APPROVED** [review]
LGTM :)

**Review by johngwilkinson — APPROVED** [review]

**Pull request merged**